apps/aetx/src/aetx_env.erl

```diff
@@ -56,40 +69,69 @@
 %%% API
 %%%===================================================================
 
--spec contract_env(aec_blocks:height(), non_neg_integer()) -> env().
-contract_env(Height, ConsensusVersion) ->
-    #env{ consensus_version = ConsensusVersion
+-spec contract_env(aec_blocks:height(), non_neg_integer(),
+                   non_neg_integer(), aec_keys:pubkey(),
+                   aec_pow:difficulty()
+                  ) -> env().
+contract_env(Height, ConsensusVersion, Time, Beneficiary, Difficulty) ->
+    #env{ beneficiary = Beneficiary
+        , consensus_version = ConsensusVersion
         , context = aetx_contract
+        , difficulty = Difficulty
         , height  = Height
         , signed_tx = none
+        , time = Time
      }.
 
--spec tx_env(aec_blocks:height(), non_neg_integer()) -> env().
-tx_env(Height, ConsensusVersion) ->
-    #env{ consensus_version = ConsensusVersion
+-spec tx_env_from_key_header(aec_headers:key_header(),
+                             aec_blocks:block_header_hash(),
+                             non_neg_integer(),
+                             aec_blocks:block_header_hash()) ->
+                             env().
+tx_env_from_key_header(KeyHeader,_KeyHash, Time,_PrevHash) ->
+    %% TODO: Add more stuff to the env
+    #env{ beneficiary = aec_headers:beneficiary(KeyHeader)
+        , consensus_version = aec_headers:version(KeyHeader)
         , context = aetx_transaction
-        , height  = Height
+        , difficulty = aec_headers:difficulty(KeyHeader)
```

Comment:
I see this in scope of ["Set the difficulty in contract dispatch"](https://www.pivotaltracker.com/n/projects/2159793/stories/159522571).
I see this means `DIFFICULTY` returns value proportional to expected time to find a solution. Ok.
I believe a note shall be added in spec (protocol repo - [here](https://github.com/aeternity/protocol/blob/108d94fc543c79b78a97608d512bba3963f6c6c5/contracts/aevm.md#chain-specific-instructions)?) mentioning:
* It is the difficulty of the previous key block of the current microblock the tx is being processed in.
  * I am considering that users may expect this to rather be difficulty of *next* keyblock header being mined.
* The format of this integer. This was not under consensus before hence the new need to document.
* What value this has in state channel, or ensure there is a Pivotal ticket tracking this.

Reply:
Partly described in https://github.com/aeternity/protocol/pull/209
The format of the integer is not yet described.